8050: JMC Automated Analysis should improve the messages when ignoring to evaluate some rules.

Suchitainf opened the pull request:
One of our customers requested to see a detailed _Ignore Reason_ whenever we ignore any rule in JMC. Rather than just saying Ignored. As part of this PR, we have added a new field Ignored Reason which will be displayed for all the Ignored Rules. It will give a clear idea about the missing event availability requirement.

The Automated Analysis screen will look like this:
![image](https://user-images.githubusercontent.com/11155712/233875905-f70cf844-e9b3-405d-a9be-12a169abfaf1.png)

In the above screenshot, there are few rules which are Ignored and the corresponding reasons are mentioned.

Please review the PR and let me know your review comments.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Issue
 * [JMC-8050](https://bugs.openjdk.org/browse/JMC-8050): JMC Automated Analysis should improve the messages when ignoring to evaluate some rules.


### Reviewers
 * [Marcus Hirt](https://openjdk.org/census#hirt) (@thegreystone - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jmc.git pull/481/head:pull/481` \
`$ git checkout pull/481`

Update a local copy of the PR: \
`$ git checkout pull/481` \
`$ git pull https://git.openjdk.org/jmc.git pull/481/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 481`

View PR using the GUI difftool: \
`$ git pr show -t 481`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jmc/pull/481.diff">https://git.openjdk.org/jmc/pull/481.diff</a>

</details>


### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jmc/pull/481#issuecomment-1519224270)

thegreystone commented:
Hi @Suchitainf! I'm thinking the single line of "Ignored" is eating valuable screen estate, and that it forces you to expand the rule result to see if a result was ignored. Should we perhaps have an "Ignored" result that renders as Ignored instead of N/A?

thegreystone commented:
Also, it would perhaps be less confusing if there wasn't an implicit double negation when presenting the results. Instead of saying: "The rule was not evaluated because the following requirements were not met: X - AVAILABLE, Y - ENABLED." Perhaps instead say, The rule could not be evaluated because: No events available of X, The Y event was disabled."

Now, the first time I read it, I read it as X was available, and I was momentarily confused as to it being available would be a problem.

Suchitainf commented:
> Hi @Suchitainf! I'm thinking the single line of "Ignored" is eating valuable screen estate, and that it forces you to expand the rule result to see if a result was ignored. Should we perhaps have an "Ignored" result that renders as Ignored instead of N/A?

Yes we can do that.

Suchitainf commented:
Sounds good. But in the actual JFRs there are 5 values of status possible and we will have to create several meaningful messages based on permutation and combinations.
These are the 5 possible values with their meanings:
AVAILABLE(4) : The type has events available in the collection.
ENABLED(3) : The type was actively enabled in the collection.
DISABLED(2) : The type was actively disabled in the collection.
NONE(1) : The type is known in the collection, but no events were found.
UNKNOWN(0) : The type is unknown in the collection.

Suchitainf commented:
Implemented all the review comments.

What's new?
1. Introduced a new severity "Ignore" apart from Info, Warning and NA.
2. All the ignored rules will be displayed with a label "Ignored". So that the user is clear that this rule is ignored.
3. Expand ignored rule to see more info. If user is interested in ignore reason for any particular rule, he/she can expand the rule to see all the reasons for ignoring that rule.
4. Introduced a new checkbox at top (Next to Show OK results), Show IGNORED results. The visibility of the ignored rules can be toggled by clicking that button. By default ignored rules wont be visible.

Few screenshots for better understanding:

Default screen:
![image](https://github.com/openjdk/jmc/assets/11155712/745e6d31-7e42-4f83-be04-cc7ef59e9aa1)

Show IGNORED Results button placement:
![image](https://github.com/openjdk/jmc/assets/11155712/ac07d1f5-5393-4fab-aa37-149242ff2056)

Click on Show Ignored icon:
![image](https://github.com/openjdk/jmc/assets/11155712/c21615e4-1086-421d-96d3-ed1bf2c59bf8)

Click on Show OK icon also:
![image](https://github.com/openjdk/jmc/assets/11155712/4af59a82-2d9b-478d-a988-676d6173867d)

Uncheck Show Ignore Icon:
![image](https://github.com/openjdk/jmc/assets/11155712/20fc68d2-c3ee-44e6-9abe-d4a517a3d73b)

Description when rule is ignored because of 2 reasons:
![image](https://github.com/openjdk/jmc/assets/11155712/e0ea2401-5a42-4960-8149-c221c54886b1)

Description when rule is ignored because of 1 reason:
![image](https://github.com/openjdk/jmc/assets/11155712/d0525b71-29f4-4e09-80b2-be772e8096fc)

Description when rule is ignored because of 4 reasons:
![image](https://github.com/openjdk/jmc/assets/11155712/3b17b7c2-765b-4a0f-824b-ab374b099294)

thegreystone commented:
Looks good! Thank you for your contribution @Suchitainf!

Suchitainf commented:
/integrate
